SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h

```diff
@@ -21,7 +23,8 @@ struct FConnectionConfig
 		, EnableWorkerSDKOpLogging(false)
 		, WorkerSDKLogFileSize(10 * 1024 * 1024)
 		, WorkerSDKLogLevel(WORKER_LOG_LEVEL_INFO)
-		, LinkProtocol(WORKER_NETWORK_CONNECTION_TYPE_KCP)
+		, LinkProtocol(WORKER_NETWORK_CONNECTION_TYPE_TCP)
+		, OverrideLinkProtocol(false)
```

Comment:
I have suggestion on improving this usage of the override bool, I think this can be done in a way that is less stateful and a bit easier to maintain reason about:
`struct FConnectionConfig
{
	enum class WorkerType
	{
		Client,
		Server
	}

	void PreConnectInit(const WorkerType Type)
	{
		//
		LinkProtocol = ConntectionTypeMap[Type];
	}

	void GetLinkProtocol(const TCHAR* CommandLine)
	{
		FString LinkProtocolString;
		FParse::Value(CommandLine, TEXT("linkProtocol"), LinkProtocolString);
		if (LinkProtocolString.Compare(TEXT("Tcp"), ESearchCase::IgnoreCase) == 0)
		{
			ConntectionTypeMap.Add(WorkerType::Client, WORKER_NETWORK_CONNECTION_TYPE_TCP);
			ConntectionTypeMap.Add(WorkerType::Server, WORKER_NETWORK_CONNECTION_TYPE_TCP);
			return;
		}
		else if (LinkProtocolString.Compare(TEXT("Kcp"), ESearchCase::IgnoreCase) == 0)
		{
			ConntectionTypeMap.Add(WorkerType::Client, WORKER_NETWORK_CONNECTION_TYPE_KCP);
			ConntectionTypeMap.Add(WorkerType::Server, WORKER_NETWORK_CONNECTION_TYPE_KCP);
			return;
		}

		if (!LinkProtocolString.IsEmpty())
		{
			UE_LOG(LogConnectionConfig, Warning, TEXT("Unknown network protocol '%s' specified for connecting to SpatialOS."),
				   *LinkProtocolString);
		}

		ConntectionTypeMap.Add(WorkerType::Client, WORKER_NETWORK_CONNECTION_TYPE_TCP);
		ConntectionTypeMap.Add(WorkerType::Server, WORKER_NETWORK_CONNECTION_TYPE_KCP);

		UE_LOG(LogConnectionConfig, Verbose, TEXT("No link protocol set. Defaulting to TCP for server workers, KCP for client workers."));
	}

private:
	TMap<WorkerType, Worker_NetworkConnectionType> ConntectionTypeMap;
}`

Reply:
Nice suggestion, added it in now (and resolving this comment soon™️)